migration/FromMySqlToPostgreSql/FromMySqlToPostgreSql.php

```diff
@@ -647,24 +659,30 @@ private function populateTableWorker(
                 }
 
                 if ($boolValidCsvEntity) {
-                    fwrite($resourceCsv, implode("\t", $arrSanitizedCsvData) . "\n");
+                    $copyArray[] = implode("\t", $arrSanitizedCsvData) . "\n";
                 }
 
-                unset($arrRow, $arrSanitizedCsvData, $boolValidCsvEntity);
-            }
-
-            // Copy current chunk into database.
-            $sqlCopy   = "COPY \"" . $this->strSchema . "\".\"" . $strTableName . "\" FROM '" . $strAddrCsv . "' WITH (FORMAT text);";
-            $stmt      = $this->pgsql->query($sqlCopy);
-            $intRetVal = count($stmt->fetchAll(\PDO::FETCH_ASSOC));
-            unset($stmt);
-            $this->log(
-                "\t-- For now inserted: " . ($intForNowInserted + $intRetVal) . ' rows, '
-                . 'Total rows in "' . $this->strSchema . '"."' . $strTableName . '": ' . $intRowsCnt . PHP_EOL
-            );
-
-            if ($intRowsCnt != 0 && 0 == $intRetVal) {
-                $this->log("\t--Following MySQL query will return a data set, rejected by PostgreSQL:\n" . $sql . "\n");
+                if ($copiedCount % $intRowsInChunk) {
+                    // Attempt copy, if it failes, do one at a time to ensure we find all the errors.
+                    // If the data is valid, we perform fast, otherwise we ensure correctness at the cost of speed.
+
```

Comment:
In my humble experience, in most cases when copy fails, it is not due to some error in one single record, so if the script will perform one insertion per one record, likely it will result in
cascading failures. It will be very slow process.  
Even if the problem is only in one single record, which is very unlikely, inserting each record in a separate transaction is very slow.